src/parser/monk/mistweaver/modules/features/AlwaysBeCasting.tsx

```diff
@@ -22,6 +24,47 @@ class AlwaysBeCasting extends CoreAlwaysBeCastingHealing {
     SPELLS.INVOKE_YULON_THE_JADE_SERPENT.id,
   ];
 
+  constructor(options: Options) {
+    super(options);
+    if(this.selectedCombatant.hasTalent(SPELLS.RISING_MIST_TALENT)) {
+      AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.RISING_SUN_KICK.id);
+      AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.RISING_SUN_KICK_SECOND.id);
+    }
+    if(this.selectedCombatant.hasTalent(SPELLS.SONG_OF_CHI_JI_TALENT)) {
+      this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.INVOKE_CHI_JI_THE_RED_CRANE_TALENT), this.handleChijiStart);
+      this.addEventListener(Events.death.to(SELECTED_PLAYER_PET), this.handleChijiDeath);
+    }
+    if(this.selectedCombatant.hasLegendaryByBonusID(SPELLS.ANCIENT_TEACHINGS_OF_THE_MONASTERY.bonusID)){
+      AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.TIGER_PALM.id);
+      AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.RISING_SUN_KICK.id);
+      AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.RISING_SUN_KICK_SECOND.id);
+      AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.BLACKOUT_KICK.id);
+    }
+
+  }
+
+  handleChijiStart(event: CastEvent) {
+    AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.TIGER_PALM.id);
+    AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.RISING_SUN_KICK.id);
+    AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.RISING_SUN_KICK_SECOND.id);
+    AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.push(SPELLS.BLACKOUT_KICK.id);
+  }
+
+  handleChijiDeath(event: DeathEvent) {
+    const tpSpot = AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.indexOf(SPELLS.TIGER_PALM.id);
+    const rskSpot = AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.indexOf(SPELLS.RISING_SUN_KICK.id);
+    const rskTwoSpot = AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.indexOf(SPELLS.RISING_SUN_KICK_SECOND.id);
+    const bokSpot = AlwaysBeCasting.HEALING_ABILITIES_ON_GCD.indexOf(SPELLS.BLACKOUT_KICK.id);
+    //if tp isn't there then we don't want to mess with this
+    if(tpSpot === -1){
+      return;
+    }
+    delete AlwaysBeCasting.HEALING_ABILITIES_ON_GCD[tpSpot];
+    delete AlwaysBeCasting.HEALING_ABILITIES_ON_GCD[rskSpot];
+    delete AlwaysBeCasting.HEALING_ABILITIES_ON_GCD[rskTwoSpot];
+    delete AlwaysBeCasting.HEALING_ABILITIES_ON_GCD[bokSpot];
+  }
+
```

Comment:
interesting approach to this - does it actually the result you want with a decreased ABC value for those 4spells?

Reply:
yes